### PR TITLE
Update grammar

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/grammar.txt [hunk 1]
```diff
@@ -180,6 +184,7 @@ AssemblyCase = 'case' Literal AssemblyBlock
 AssemblyDefault = 'default' AssemblyBlock
 AssemblyForLoop = 'for' AssemblyBlock AssemblyExpression AssemblyBlock AssemblyBlock
 AssemblyBreakContinue = 'break' | 'continue'
+AssemblyLeave = 'leave'
```
Comment (axic) [inline docs/grammar.txt:187]: Did you add this to the yul grammar too (`docs/yul.rst`)?

Comment (chriseth) [reply docs/grammar.txt:187]: That's up last in the list: https://github.com/ethereum/solidity/issues/7970 ;)

File: docs/grammar.txt [hunk 2]
```diff
@@ -16,17 +16,20 @@ ContractPart = StateVariableDeclaration | UsingForDeclaration
...
 InheritanceSpecifier = UserDefinedTypeName ( '(' Expression ( ',' Expression )* ')' )?
 
-StateVariableDeclaration = TypeName ( 'public' | 'internal' | 'private' | 'constant' )* Identifier ('=' Expression)? ';'
+StateVariableDeclaration = TypeName ( 'public' | 'internal' | 'private' | 'constant' | OverrideSpecifier )* Identifier ('=' Expression)? ';'
 UsingForDeclaration = 'using' Identifier 'for' ('*' | TypeName) ';'
 StructDefinition = 'struct' Identifier '{'
                      ( VariableDeclaration ';' (VariableDeclaration ';')* ) '}'
 
-ModifierDefinition = 'modifier' Identifier ParameterList? Block
+ModifierDefinition = 'modifier' Identifier ParameterList?  ( 'virtual' | OverrideSpecifier? )* Block
```
Comment (leonardoalt) [inline docs/grammar.txt]: Why `?`?

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**